ui/src/app.js

```diff
@@ -1509,14 +1509,12 @@ function getUsedCassettes (cassettes) {
 
   // This function only serves to create an array mapping what buttons are shown
   // More UI buttons should allow for simpler solutions to this issue
-  const doesCassette3HaveBills = cassettes.length >= 4 && cassettes[2].count > 0
   const doesCassette4HaveBills = cassettes.length >= 5 && cassettes[3].count > 0
-  const thirdCassette =
-    doesCassette3HaveBills
-      ? cassettes[2]
-      : doesCassette4HaveBills
-        ? cassettes[3]
-        : cassettes[4]
+  // cassette[2] is the virtual cassette in 2-cassette mode and the third cassette on 3-cassette mode
+  const thirdCassette = 
+    doesCassette4HaveBills
+      ? cassettes[3]
+      : cassettes[2]
   return [cassettes[0], cassettes[1], thirdCassette]
```

Comment:
If the (physical) cassettes 3 and 4 of a 4-cassettes machine are empty, doesn't this put the (empty) cassette 4 as the third, when the virtual cassette could be used instead?

If I'm not misunderstanding the problem, wouldn't it be easier to filter and then take (up to) 3 cassettes off the result? `take 3 . filter (\c -> c.count > 0)`, or `_.flow(_.filter(c => c.count > 0), _.take(3))(cassettes)`, or `cassettes.filter(c => c.count > 0).slice(0, 3)` ?